### ISSUE TITLE
StyleCLIP: Text-Driven Manipulation of StyleGAN Imagery

### ISSUE THREAD

**Issue by ghuijo** [One-Day-One-Paper/One-Day-One-Paper#74]
*"StyleGAN2과 CLIP의 결합"*
*"Image manipulation 관련 논문"*

#### 처음
* CLIP: Contrastive Language-Image Pre-training
  * pretrained on 400 million image-text pairs
* StyleGAN과 CLIP을 결합하는 세가지 방법 제안
  * Text-guided latent optimization, where a CLIP model is used as a loss network
  * A latent residual mapper, trained for a specific text prompt
  * mapping a text prompt into an inputagnostic (global) direction in StyleGAN’s style space,
providing control over the manipulation strength as well as the degree of disentanglement

#### StyleCLIP Text-Driven Manipulation
* Latent Optimization
  * "direct latent code optimization"
  * <image src="https://user-images.githubusercontent.com/67627471/152684450-2836922a-3ba8-4aa6-8948-053845ac0dd4.png" width="40%" height="40%" >
  * G는 사전학습된 StyleGAN의 generator이고 D_CLIP은 이미지와 텍스트간의 CLIP 임베딩 거리
  * latent image embedding w / source latent code w_s (둘 다 W+ space 내)
  * <image src="https://user-images.githubusercontent.com/67627471/152684467-2ad9c7f9-6203-4e28-ab34-e21b07bf0b4b.png" width="30%" height="30%" >
  * R: pretrained ArcFace network for face recognition
  * λ_L2와 λ_ID 값에 따른 결과
  <image src="https://user-images.githubusercontent.com/67627471/152684906-4f4eb8a2-a6ee-4109-b85a-98ac3a3b7cc7.png" width="50%" height="50%" >
* Latent Mapper
  * <image src="https://user-images.githubusercontent.com/67627471/152685041-b0e1d6af-863e-45f0-8e42-71d534abd7a8.png" width="30%" height="30%" >
  * 여기서, w = ( w_c, w_m, w_f )
  * CLIP loss
    * CLIP(w) guides the mapper to minimize the cosine distance in the CLIP latent space
    * <image src="https://user-images.githubusercontent.com/67627471/152685146-4de6ab77-232c-41e7-815f-d213fb6fe272.png" width="30%" height="30%" >
  * total loss function
    * <image src="https://user-images.githubusercontent.com/67627471/152685170-21c50617-7f27-4733-8038-71047dc3feb9.png" width="35%" height="35%" >
  * text-guided mapper 구조
  <image src="https://user-images.githubusercontent.com/67627471/152685276-c004dac9-f303-469c-9d01-fecda3efb317.png" width="60%" height="60%" >

  * 결과
  <image src="https://user-images.githubusercontent.com/67627471/152685370-c6adca65-964f-4a3c-b4f0-a2823f47246e.png" width="50%" height="50%" >

  * 문제점
    * sometimes falls short when a fine-grained disentangled manipulation is desired -> 아래에서 해결
* Global Directions
  * global latent space에 text mapping
  * a pair of images; G(s), G(s+α∆s)
  * difference between the two images in CLIP space is given by ∆i
  * Given a natural language instruction encoded as ∆t, and assuming collinearity between ∆t and ∆i,
  determine a manipulation direction ∆s by assessing the relevance of each channel in S to the direction ∆i
  * Channelwise relevance
    * <image src="https://user-images.githubusercontent.com/67627471/152685709-b57148d4-5a84-49fd-b7fb-65406b3e86b8.png" width="30%" height="30%" >
    * <image src="https://user-images.githubusercontent.com/67627471/152685773-6f9841a3-9c30-42b9-b1e7-c18dc0dd8c2d.png" width="50%" height="50%">
    * 임계점 알파 커지면 효과 세짐 / 임계점 베타가 커지면 머리카락 색만 영향 받고 낮아지면 주름 등 얼굴 요소도 영향 받음
    * 여기서 임계점은 위에 있는 R_c값의 임계점 말함

#### 결과
* TediGAN과 제안한 2가지 방법 비교
  <image src="https://user-images.githubusercontent.com/67627471/152685942-8dfba68c-2a5a-4735-a02e-835ed89ccc74.png" width="80%" height="80%">
  * Global이 적절한 사진을 만들어냄. Mapper는 값 조절이 안 돼 너무 극단적인 결과가 나오는 듯..

##
* StyleGAN 활용을 이렇게 할 수 있구나..
* 클립 / StyleGAN3(Alias-Free GAN) 공부
* Global Directions 부분 어려움
